unified_planning/model/timing.py

```diff
@@ -28,28 +28,49 @@ class TimepointKind(Enum):
 
 
 class Timepoint:
-    def __init__(self, kind: TimepointKind):
+    def __init__(self, kind: TimepointKind, container=None):
```

Comment:
Missining `container` type annotation.

Reply:
Yes, my understanding is that we cannot provide the type annotation here because it would create a circular dependency: `Subtask` uses `Timepoint` for its `start`/`end` properties so referencing the `Subtask` class here would create create a cycle in the imports.
So I relied on duck-typing and method documentation.

Reply:
To avoid the circular dependency, you can specify the type annotation as a string.

```suggestion
    def __init__(self, kind: TimepointKind, container: Optional['Subtask'] = None):
```

Reply:
I already tried that without success :/ Could you give a try ? I might be missing some subtleties on how mypy works.

Reply:
You are right, it's harder than I thought. You can ignore this comment for the moment.

Reply:
Ok, I removed the annotation and tests are now passing again.